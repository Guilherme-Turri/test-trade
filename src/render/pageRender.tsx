import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../styles/theme';
import store from '../storage/store';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

export const pageRender = (children: React.ReactNode) => {
  return render(
    <ThemeProvider theme={theme}>
      <Provider store={store}>
        <BrowserRouter>
          {children}
        </BrowserRouter>
      </Provider>
    </ThemeProvider>);
};