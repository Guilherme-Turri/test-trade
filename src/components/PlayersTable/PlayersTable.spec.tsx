import { componentRender } from "../../render/componentRender";
import { screen } from '@testing-library/react'
import { PlayersTable } from ".";

describe('<PlayersTable/>', () => {
  const mockPlayers = [
    {
      player: {
        name: 'playerName1',
        age: 25,
        nationality: 'Brazil',
      },
    }
  ];

  it('should render on screen properly', () => {
    const { container } = componentRender(<PlayersTable players={mockPlayers} />)
    const table = screen.getByTestId('playersTable')
    const playerName = screen.getByText('playerName1')
    expect(table).toBeInTheDocument()
    expect(playerName).toBeInTheDocument()
    expect(container).toMatchSnapshot();
  });

})
