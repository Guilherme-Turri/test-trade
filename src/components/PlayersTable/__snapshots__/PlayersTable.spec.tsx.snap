// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlayersTable/> should render on screen properly 1`] = `
.c1 {
  text-transform: none;
  color: #00c555;
  font-size: 20px;
  font-family: 'Bruno Ace SC',cursive;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100%;
  padding: 0.938rem;
  margin: 0.938rem;
}

.c0 th {
  padding: 0.938rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: white;
}

.c0 td {
  text-align: center;
  padding: 0.938rem;
  background-color: white;
}

.c0 caption {
  font-weight: bold;
  font-size: 1.5rem;
}

.c0 table {
  margin-top: 0.938rem;
  width: 100%;
}

<div>
  <div
    class="c0"
    data-testid="playersTable"
  >
    <h3
      class="c1"
    >
      Lista de Jogadores
    </h3>
    <table>
      <thead>
        <tr>
          <th>
            Nome
          </th>
          <th>
            Idade
          </th>
          <th>
            Nacionalidade
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            playerName1
          </td>
          <td>
            25
          </td>
          <td>
            Brazil
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
